storage/filesystem/object.go

```diff
@@ -28,23 +28,22 @@ type ObjectStorage struct {
 }
 
 // NewObjectStorage creates a new ObjectStorage with the given .git directory.
-func NewObjectStorage(dir *dotgit.DotGit) (ObjectStorage, error) {
-	return NewObjectStorageWithOptions(dir, Options{})
-}
-
-// NewObjectStorageWithOptions creates a new ObjectStorage with the given .git
-// directory and sets its options.
-func NewObjectStorageWithOptions(
-	dir *dotgit.DotGit,
-	ops Options,
-) (ObjectStorage, error) {
-	s := ObjectStorage{
-		options:        ops,
+func NewObjectStorage(dir *dotgit.DotGit) *ObjectStorage {
+	return &ObjectStorage{
+		options:        Options{},
 		deltaBaseCache: cache.NewObjectLRUDefault(),
 		dir:            dir,
 	}
+}
+
+func (s *ObjectStorage) WithOptions(ops Options) *ObjectStorage {
+	s.options = ops
+	return s
+}
 
-	return s, nil
+func (s *ObjectStorage) WithCache(cache cache.Object) *ObjectStorage {
+	s.deltaBaseCache = cache
```

Comment:
The cache should be passed to the constructor in the same way we do with the dotgit instance since is required, the options are optional, but if in the case of the cache the LRU is being instanced always in the constructor wasting resources.


`func NewObjectStorage(dir *dotgit.DotGit, cache cache.Object) *ObjectStorage {`